status.c

```diff
@@ -56,6 +58,45 @@ extern void setpipestatus(int stats[], int num) {
 	}
 }
 
+/* wait on multiple processes and store their exit status */
+
+extern void setwaitstatus(char **av, char *cmd) {
+	int i, j, count;
+	pid_t pid;
+
+	for (count = 0; av[count] != NULL; count++)
+		;
+	
+	/* ensure we have enough space to store all the results */
+	if (count >= sizeof(statuses)) {
+		fprint(2, RC "too many arguments to %s\n", cmd);
+		set(FALSE);
+		return;
+	}
+
+	/* we need to fill statuses backwards */
```

Comment:
Why do we need to fill statuses backwards?

Reply:
I had to do this because otherwise you get the status of the last executed command first in `$status`. I did not look further why this is the case.